tests/test_diffraction.py

```diff
@@ -0,0 +1,51 @@
+import freud
+import matplotlib
+import unittest
+import numpy.testing as npt
+matplotlib.use('agg')
+
+
+class TestDiffractionPattern(unittest.TestCase):
```

Comment:
Is there a limiting case of the diffraction pattern or some kind of pattern that we already know, so that we can have this compute it and compare answers?

Reply:
Consider zero particles, one particle, etc. The center value of the array is S(q=0) which should be the highest value in the image (for images with even side lengths, it may be a cluster of 4 pixels in the center).

Reply:
I won't be able to make these changes for a little while -- @tommy-waltmann, feel free to give it an attempt.

Reply:
Another potential method is to convert the diffraction back into particle positions with iFFT + converting the everything back into discrete points. I've done this before while playing with the diffraction module if this is something we'd be interested in.

Reply:
You could do iFFT on the diffraction and then descritize those points and compare them to the simulation points? I'm not sure if that's an appropriate test to see if diffraction worked, but I've done that before with my own simulations while playing with the diffraction module

Reply:
Is this basically the inverse operation of the diffraction? Any cross-checks on these calculations are useful so feel free to add that to the tests.

Reply:
Yup. I'll add it in

Reply:
@klywang Just to clarify, the user gets `S(q)` out of this calculation, not `I(q)`<sup>1</sup>. There is no complex phase associated with the resulting array that could be used to take the inverse FFT and recover raw point locations (unless there's some magic trick here that I don't know). Regardless, there are plenty of other helpful tests that can be written! 😄

<sup>1</sup> Definitions on Wikipedia: https://en.wikipedia.org/wiki/Structure_factor

Reply:
Ohh. There was no magic -- I forgot that I did that while looking at the source code and had access to the imaginary portions.

Reply:
I've started work on these, and the tests for the max value at the center are failing. Also, is there a way to get freud to accept an empty system?

Reply:
Not that I'm aware of, the NeighborQuery construction will fail IIRC, but you might be able to spoof that behavior by feeding an empty NeighborList (or equivalently, an r_max of 0).

Reply:
So I've looked into why the tests for the center position are failing, and it is because the actual value at the center pixel is not the max, although the region in the center is always a maximum. I propose we alter the tests to test whether the center pixel has a value the same order of magnitude as the maximum value. Does that sound good to everyone else?

Reply:
@tommy-waltmann So you're saying there's a higher peak at some k-vector that is not zero? This is plausible because the center pixels may share the intensity, e.g. the center 4 pixels might have `I(k=0)/4` each. It might be worth testing this hypothesis, you can sum over 3x3 regions (or maybe 2x2 regions) and verify if the highest sum is in the center. I _think_ that should always be true.

Reply:
Basically, there is a region in the center which is a maximum. Of the pixels in that region in the center, the central one is not quite as large as some nearby ones, but it is larger than areas not in the center.

Reply:
I've updated the tests to test for the existence of the peak at the center. We previously mentioned wanting to do some validation of a specific limiting case, like 1 particle as @bdice mentioned earlier. Is that something we could explicitly compute via another method by hand, or what is the best way to validate that case?

Reply:
I've added a test for a single particle system that mimics a 0 particle system. There is a single point at the origin, and a large box of length 100.